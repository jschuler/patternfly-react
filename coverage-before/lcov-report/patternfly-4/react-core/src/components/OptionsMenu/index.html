<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for patternfly-4/react-core/src/components/OptionsMenu</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> patternfly-4/react-core/src/components/OptionsMenu
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">96.84% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>92/95</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">58.25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>60/103</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.29% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">96.81% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>91/94</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="OptionsMenu.tsx"><a href="OptionsMenu.tsx.html">OptionsMenu.tsx</a></td>
	<td data-value="95.24" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="21" class="abs high">20/21</td>
	<td data-value="64.29" class="pct medium">64.29%</td>
	<td data-value="14" class="abs medium">9/14</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="21" class="abs high">20/21</td>
	</tr>

<tr>
	<td class="file high" data-value="OptionsMenuItem.tsx"><a href="OptionsMenuItem.tsx.html">OptionsMenuItem.tsx</a></td>
	<td data-value="92.86" class="pic high"><div class="chart"><div class="cover-fill" style="width: 92%;"></div><div class="cover-empty" style="width:8%;"></div></div></td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="14" class="abs high">13/14</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="12" class="abs medium">7/12</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="14" class="abs high">13/14</td>
	</tr>

<tr>
	<td class="file high" data-value="OptionsMenuItemGroup.tsx"><a href="OptionsMenuItemGroup.tsx.html">OptionsMenuItemGroup.tsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="14" class="abs medium">10/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file high" data-value="OptionsMenuSeparator.tsx"><a href="OptionsMenuSeparator.tsx.html">OptionsMenuSeparator.tsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="OptionsMenuToggle.tsx"><a href="OptionsMenuToggle.tsx.html">OptionsMenuToggle.tsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="58.62" class="pct medium">58.62%</td>
	<td data-value="29" class="abs medium">17/29</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="OptionsMenuToggleWithText.tsx"><a href="OptionsMenuToggleWithText.tsx.html">OptionsMenuToggleWithText.tsx</a></td>
	<td data-value="95.65" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95.65" class="pct high">95.65%</td>
	<td data-value="23" class="abs high">22/23</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="32" class="abs medium">16/32</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="95.65" class="pct high">95.65%</td>
	<td data-value="23" class="abs high">22/23</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jan 17 2020 19:28:45 GMT-0500 (Eastern Standard Time)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
<script src="../../../../../block-navigation.js"></script>
</body>
</html>
