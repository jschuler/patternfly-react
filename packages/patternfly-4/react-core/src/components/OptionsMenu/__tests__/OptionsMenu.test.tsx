// Auto-generated do not edit

import * as React from 'react';
import { mount } from 'enzyme';
import { OptionsMenu } from '../OptionsMenu';

describe('OptionsMenu test', () => {
  it('OptionsMenu should match snapshot', () => {
    const view = mount(
      <OptionsMenu
        className={"''"}
				id={"'defaultString'"}
				menuItems={unrecognizedType Array undefined, consider reporting error to react-jest-test-generator.}
				toggle={<p>ReactElement</p>}
				isPlain={undefined}
				isOpen={undefined}
				isText={false}
				isGrouped={false}
				ariaLabelMenu={undefined}
				position={undefined}
				direction={undefined}
      />);
    expect(view).toMatchSnapshot();
  });
});
