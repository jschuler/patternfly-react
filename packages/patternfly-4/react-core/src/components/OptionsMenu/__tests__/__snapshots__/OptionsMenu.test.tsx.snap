// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsMenu test OptionsMenu should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "baseClass": "pf-c-options-menu",
      "baseComponent": "div",
      "disabledClass": "pf-m-disabled",
      "id": "'defaultString'",
      "itemClass": "pf-c-options-menu__menu-item",
      "menuClass": "pf-c-options-menu__menu",
      "menuComponent": "ul",
      "onSelect": [Function],
      "toggleClass": "pf-c-options-menu__toggle",
      "toggleIconClass": "pf-c-options-menu__toggle-icon",
      "toggleTextClass": "pf-c-options-menu__toggle-text",
    }
  }
>
  <Component
    className="''"
    dropdownItems={Array []}
    id="'defaultString'"
    isGrouped={false}
    toggle={
      <p>
        ReactElement
      </p>
    }
  />
</ContextProvider>
`;
